bin/node/cli/src/service.rs

```diff
@@ -319,7 +319,13 @@ pub fn new_full_base(
 							slot_duration,
 						);
 
-					Ok((timestamp, slot, uncles))
+					let storage_proof =
+						sp_transaction_storage_proof::registration::new_data_provider(
+							&*client_clone,
+							&parent,
+						)?;
+
+					Ok((timestamp, slot, uncles, storage_proof))
```

Comment:
I don't really understand how there was no error for this before (we did change the inherentsdataprovider type here without other changes in the pr)?
Or is it related to async_trait doing some ~ stuf?

Reply:
What you mean? Not every inherent data is required, meaning the runtime doesn't panic when a certain inherent data isn't provided.

Reply:
I mean on the rust type level (from a tree item tuple to a four one).

Reply:
Right, no proof is required if there were no indexed transactions in the block

Reply:
> I mean on the rust type level (from a three item tuple to a four one).

You can return there (theoretically) as many tuple elements as you like. The number is not enforced anywhere.

Reply:
Yes it is indeed a feature from async_trait crate. Similar to how async does.

Reply:
No. Nothing of that is related to `async_trait`.

Reply:
Return type did change without any calling code.
So calling code probably use on a trait as input (I would think inherent provider might be implemented for multiple tuples).
Indeed some 'impl_trait_for_tuples' is being use.
So simple type inference (not sure if it run with async or async_trait does something, but will trust you on this one)..